tests/models/device.spec.coffee

```diff
@@ -14,276 +13,180 @@ describe 'Device Model:', ->
-					@pineGetStub.returns(Promise.resolve([ @device ]))
+			afterEach ->
+				@pineGetStub.restore()
 
-				afterEach ->
-					@pineGetStub.restore()
+			it 'should return the correct number of devices', (done) ->
+				device.getAllByApplication 'MyApp', (error, devices) ->
+					expect(error).to.not.exist
+					expect(devices).to.have.length(1)
+					done()
 
-				it 'should return the device', (done) ->
-					device.get 'MyDevice', (error, device) =>
-						expect(error).to.not.exist
-						expect(device).to.deep.equal(@device)
-						done()
+			it 'should add application_name', (done) ->
+				device.getAllByApplication 'MyApp', (error, devices) =>
+					expect(devices[0].application_name).to.equal(@device.application[0].app_name)
+					done()
 
-				it 'should add application_name', (done) ->
-					device.get 'MyDevice', (error, device) =>
-						expect(device.application_name).to.equal(@device.application[0].app_name)
-						done()
+	describe '.get()', ->
 
-		describe 'given no logged in user', ->
+		describe 'given no devices', ->
 
 			beforeEach ->
-				@tokenGetUsernameStub = sinon.stub(token, 'getUsername')
-				@tokenGetUsernameStub.returns(undefined)
+				@pineGetStub = sinon.stub(pine, 'get')
+				@pineGetStub.returns(Promise.resolve([]))
 
 			afterEach ->
-				@tokenGetUsernameStub.restore()
+				@pineGetStub.restore()
 
 			it 'should return an error', (done) ->
-				device.get 'MyDevice', (error, devices) ->
-					expect(error).to.be.an.instanceof(errors.ResinNotLoggedIn)
-					expect(devices).to.not.exist
+				device.get 'MyDevice', (error, device) ->
+					expect(error).to.be.an.instanceof(errors.ResinDeviceNotFound)
+					expect(device).to.not.exist
 					done()
 
-	describe '.has()', ->
-
-		describe 'given a logged in user', ->
+		describe 'given a device', ->
 
 			beforeEach ->
-				@tokenGetUsernameStub = sinon.stub(token, 'getUsername')
-				@tokenGetUsernameStub.returns('johndoe')
-
-			afterEach ->
-				@tokenGetUsernameStub.restore()
+				@device =
+					id: 1
+					name: 'Device1'
+					application: [
+						app_name: 'App1'
+					]
 
-			describe 'given no devices', ->
+				@pineGetStub = sinon.stub(pine, 'get')
+				@pineGetStub.returns(Promise.resolve([ @device ]))
 
-				beforeEach ->
-					@pineGetStub = sinon.stub(pine, 'get')
-					@pineGetStub.returns(Promise.resolve([]))
+			afterEach ->
+				@pineGetStub.restore()
 
-				afterEach ->
-					@pineGetStub.restore()
+			it 'should return the device', (done) ->
+				device.get 'MyDevice', (error, device) =>
+					expect(error).to.not.exist
+					expect(device).to.deep.equal(@device)
+					done()
 
-				it 'should return false', (done) ->
-					device.has 'MyDevice', (error, hasDevice) ->
-						expect(error).to.not.exist
-						expect(hasDevice).to.be.false
-						done()
+			it 'should add application_name', (done) ->
+				device.get 'MyDevice', (error, device) =>
+					expect(device.application_name).to.equal(@device.application[0].app_name)
+					done()
```

Comment:
I plan to move the SDK tests to use `chai-as-promised` as the other modules, but plan to do it in another PR.